### DISCUSSION TITLE
Is it possible to customize OAuth2PasswordBearer schema name?

### DISCUSSION THREAD

**Discussion by MuslimBeibytuly** [fastapi/fastapi#8610] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
oauth2_scheme = OAuth2PasswordBearer(tokenUrl='/api/users/sign_in/')

@router.post(
    path='/users/sign_in/', status_code=200,
    response_model=SignInResponse,
)
def sign_in(
    form: OAuth2PasswordRequestForm = Depends(),
) -> SignInResponse:
```

### Description

When I open swagger at /docs, I see name of schema is: Body_sign_in_api_users_sign_in__post, I would like to rename it to have better readability

### Wanted Solution

I've no idea, I guess as an argument to request form

### Wanted Code

```python
I've no idea, I guess as an argument to request form
```

### Operating System

Linux

### Operating System Details

docker: python:3.9.10-slim

### FastAPI Version

0.74.1

### Python Version

3.9.10

**Comment by aryadovoy** [comment]
Any news?

**Comment by hurik** [comment]
Here is my solution to rename the "Body_login_auth_token_post" schema. I don't find it elegant, but unfortunately, I haven't found a better solution so far. It uses this example from the documentation: [https://fastapi.tiangolo.com/how-to/extending-openapi/#overriding-the-defaults](https://fastapi.tiangolo.com/how-to/extending-openapi/#overriding-the-defaults)

Before:
![grafik](https://github.com/user-attachments/assets/eebd7a46-f627-4585-860b-e6d9ab6c7d29)

After:
![grafik](https://github.com/user-attachments/assets/48d292e3-3fd8-4575-abe6-b72680d5cf28)

The code:
```python
import json
from typing import Annotated

from fastapi import FastAPI, Depends
from fastapi.openapi.utils import get_openapi
from fastapi.security import OAuth2PasswordRequestForm
from pydantic import BaseModel

app = FastAPI()


class AuthTokenOutMdl(BaseModel):
    access_token: str
    token_type: str


@app.post("/auth/token")
async def login(form_data: Annotated[OAuth2PasswordRequestForm, Depends()]) -> AuthTokenOutMdl:
    return AuthTokenOutMdl(access_token='XXXXX', token_type='bearer')


# https://fastapi.tiangolo.com/how-to/extending-openapi/#overriding-the-defaults
def custom_openapi():
    if app.openapi_schema:
        return app.openapi_schema

    openapi_schema = get_openapi(
        title="Example",
        version="0.1.0",
        routes=app.routes
    )

    stupid_schema_name = 'Body_login_auth_token_post'
    better_schema_name = 'AuthTokenInMdl'

    # Dict to json string, replace the schema name, and back to dict
    openapi_schema = json.loads(json.dumps(openapi_schema).replace(stupid_schema_name, better_schema_name))

    # Sort the components schemas, because new name could not be sorted correctly
    openapi_schema['components']['schemas'] = dict(sorted(openapi_schema['components']['schemas'].items()))

    app.openapi_schema = openapi_schema

    return app.openapi_schema


app.openapi = custom_openapi
```

**Comment by Vigilans** [comment]
This trick will serve both purpose of converting schema (built using \_\_init__) to pydantic model `LocalRequest` and satisfy type checker and intellisense:
```python
class LoginRequest(BaseModel, OAuth2PasswordRequestForm):
    __annotations__ = {
        name: (
            ann_type if name not in OAuth2PasswordRequestForm.__init__.__kwdefaults__
            else setattr( # type: ignore[func-returns-value]
                ann_type.__metadata__[0], "default", OAuth2PasswordRequestForm.__init__.__kwdefaults__[name]
            ) or ann_type
        )
        for name, ann_type in OAuth2PasswordRequestForm.__init__.__annotations__.items()
    }
```

It could have been more simple if not for setting the default values of some parameters:
```python
class LoginRequest(BaseModel, OAuth2PasswordRequestForm):
    __annotations__ = OAuth2PasswordRequestForm.__init__.__annotations__
```

You should `LoginRequest` in this way:
```python
@router.post("/login")
async def login(
    login_data: Annotated[LoginRequest, Form()],
    auth_service: Annotated[AuthService, Depends(get_auth_service)],
) -> LoginResponse:
```
